sdk/core/azure-core/src/main/java/com/azure/core/util/RequestContent.java

```diff
@@ -0,0 +1,288 @@
+     * @param length Length of the data.
+     * @return A new {@link RequestContent}.
+     * @throws NullPointerException If {@code file} is null.
+     * @throws IllegalArgumentException If {@code offset} or {@code length} are negative or {@code offset} plus {@code
+     * length} is greater than the file size.
+     */
+    public static RequestContent fromFile(Path file, long offset, long length) {
+        Objects.requireNonNull(file, "'file' cannot be null.");
+        if (offset < 0) {
+            throw new ClientLogger(RequestContent.class).logExceptionAsError(new IllegalArgumentException(
+                "'offset' cannot be negative."));
+        }
+        if (length < 0) {
+            throw new ClientLogger(RequestContent.class).logExceptionAsError(new IllegalArgumentException(
+                "'length' cannot be negative."));
+        }
+        if (offset + length > file.toFile().length()) {
+            throw new ClientLogger(RequestContent.class).logExceptionAsError(new IllegalArgumentException(
+                "'offset' plus 'length' cannot be greater than the file's size."));
+        }
+
+        return new FileContent(file, offset, length);
+    }
+
+    /**
+     * Creates a {@link RequestContent} that uses a serialized {@link Object} as its data.
+     * <p>
+     * This uses an {@link ObjectSerializer} found on the classpath.
+     * <p>
+     * The {@link RequestContent} returned has a null {@link #getLength()}, if the length of the content is needed use
+     * {@link BinaryData#fromObject(Object)} and {@link RequestContent#fromBinaryData(BinaryData)} to create the request
+     * content.
+     *
+     * @param serializable An {@link Object} that will be serialized to be the {@link RequestContent} data.
+     * @return A new {@link RequestContent}.
+     */
+    public static RequestContent fromObject(Object serializable) {
+        return fromObject(serializable, JsonSerializerProviders.createInstance(true));
+    }
+
+    /**
+     * Creates a {@link RequestContent} that uses a serialized {@link Object} as its data.
+     * <p>
+     * The {@link RequestContent} returned has a null {@link #getLength()}, if the length of the content is needed use
+     * {@link BinaryData#fromObject(Object, ObjectSerializer)} and {@link RequestContent#fromBinaryData(BinaryData)} to
+     * create the request content.
+     *
+     * @param serializable An {@link Object} that will be serialized to be the {@link RequestContent} data.
+     * @param serializer The {@link ObjectSerializer} that will serialize the {@link Object}.
+     * @return A new {@link RequestContent}.
+     * @throws NullPointerException If {@code serializer} is null.
+     */
+    public static RequestContent fromObject(Object serializable, ObjectSerializer serializer) {
+        Objects.requireNonNull(serializer, "'serializer' cannot be null.");
+        return new SerializableContent(serializable, serializer);
+    }
```

Comment:
`serializable`  should be non-null too.

Reply:
I'd rather leave null handling to the `ObjectSerializer`. For JSON serialization null can be an acceptable serialization value, turning into JSON null, for Avro it could result in an empty byte array.